### ISSUE TITLE
csf unabe to restart - iptables problem up-stable 3.2.1

### ISSUE THREAD

**Issue by EdNett** [omega8cc/boa#1227]
Hi,

After fixing our problems with openssh and ssh_armour we did another up-stable of 3.2.1 with ssh_armour set to NO and ssh_from_sources set to NO, php 7  only which was upgraded to 7.0.24 , and the sceeen return was that csf/lfd cannot be installed on this system ( a  scaleway vps that we use as a boa test vps).  This vps was working fine before with csf/lfd installed and running.

Gist is here:
https://gist.github.com/EdNett/a62e3c0328d56c1a7b172a178848de49

boa info more is here:
https://gist.github.com/EdNett/d53b7e15ab6ba95654e0c2bfb29630bd

Am looking into this, but it is a little out of my area of expertise. Will update.

**Comment by EdNett** [comment]
Is the problem because BOA doesn't use IPv6 and here it is trying to use ip6tables?   See the repeated error "cant initialize ip6tables filter"?

**Comment by omega8cc** [comment]
Please try to install `ip6tables` -- it should be installed with the system by default, I believe:

`apt-get install ip6tables`

Our CSF configuration expects `ip6tables` to be available, even if IPv6 is not supported in Aegir yet, because otherwise your server would be exposed to attacks and unauthorized remote access over its IPv6 address. By default csf allows connections only via port 53 TCP/UDP on IPv6.

**Comment by EdNett** [comment]
Hello,

ip6tables is not a separate program, rather it is included with iptables. I reinstalled it.

**Comment by omega8cc** [comment]
Indeed, it's just a symlink to the same binary `xtables-multi`, but the errors mentioned both ip6tables and iptables, so this has nothing to do with IPv6, and it seems to be related to your VPS kernel modules? At least, we couldn't reproduce this on any KVM guest we manage.

**Comment by EdNett** [comment]
Hello,  We just upgraded up-stable to 3.2.2, and this problem remains.  The vps provider has "group security rules" that we need to set that may be affecting this. We have set TCP to accept OUTBOUND ports 25, 53, 80, 143, and 443. 

1) Is this correct for a BOA server?
2) Do we need to add a rule to ACCEPT OUTBOUND UDP on port 53 also?
3) Do we need to accept INBOUND UDP and/or TCP on port 53, also (both ?).

Could this be the cause of our problem with the csf error on lne 1236 preventing csf from starting?

Alternatively, the kernel (re your mentioning vps kernel modules, above)  is provided by BOA isn't it? This is a vps dedicated to BOA alone. There is, however a a section in the vps configuration on the provider's website called bootscript, which is listed at x86_64 mainline 4.4.96 rev1. We have numerous choices, including x86_64 mainline 4.9.77 rev1, a 4.10.xx up to 4.12.xx and so on - is this what you are referring to? We'll try experimenting with changing this bootscript, but I don't see how it could be preventing csf from starting on a boa system.
